proposals/0000-extension-lifecycle-framework.md

```diff
@@ -0,0 +1,142 @@
+
+ * `Experimental` extensions are undergoing active development, or have consequences that make it impossible in practice to avoid breaking changes. The syntax and semantics that are enabled by the extension are likely to change regularly.
+   * It is expected that most new language extensions will begin as experimental, while the developers and the community gain experience with their use.
+   * Despite being open to breaking change, an `Experimental` extension must be `Deprecated` prior to removal.
+   * Some extensions in this category might be `RequiredTypeArguments`, `RebindableSyntax`, and `Arrows`.
+
+ * `Deprecated` extensions are considered to be design or implementation dead ends, and should not be used in new code. Deprecating an extension means that GHC may remove support for it in a future release (though this decision may be revised in the light of user feedback).
+   * When an extension is deprecated, the user's guide and the compiler warning _must_ include a statement about the longevity of the extension, though this need not necessarily commit to a concrete time for removal. Additionally, the user's guide and the warning should direct users to information about migrating away from the deprecated extension.
+   * Any extension will be deprecated prior to removal.
+   * Some extensions in this category might be `OverlappingInstances`, `Rank2Types`, `RecordPuns`, and `TypeInType`.
+
+ * `Legacy` extensions are explicitly not recommended for new code, much like `Deprecated`, however, they are expected to be supported indefinitely. This may be to maintain an older language edition, maintain a bridge to a newer feature, or otherwise keep backward compatibility. Extensions that are `Legacy` may have some conflict with `Stable` extensions and explicitly do not need to be considered for an extension to be `Stable`. Some extensions in this category might be `CUSKs`, `DeepSubsumption`,  and `NPlusKPatterns`.
+
+Above we list expected categorizations of several extensions based on statements in the user's guide. We do not attempt to prescribe any particular categorizations as part of this proposal and simply provide a small number of expectations from the perspective of someone reading the existing documentation.
+
```

Comment:
Start a new subection here: "Lifecycle: how an extension's category may change over time"

(And rename 2.1 to just "Categories"

Reply:
Done, thank you!